server/src/test/java/org/elasticsearch/index/query/SimpleQueryStringBuilderTests.java

```diff
@@ -786,4 +784,34 @@ private void assertQueryWithAllFieldsWildcard(Query query) {
         assertThat(disjunctionMaxQuery.getDisjuncts(), hasItems(new TermQuery(new Term(STRING_FIELD_NAME, "hello")),
             new TermQuery(new Term(STRING_FIELD_NAME_2, "hello"))));
     }
+
+    /**
+     * Query terms that start with "now" can trigger a query to not be cacheable if it hits all field
+     * and there is a date field amongst them. This test checks the search context cacheable flag is
+     * updated accordingly
+     * Note: this cannot happen when directly hitting the date field, dates are not parsed leniently then
+     * and we typically get a parse exception.
```

Comment:
I’m not 100% sure we need this entire test, we should only be hitting the non-caching case when we query “all” fields and are in a lenient mode.